kong/versioning.go

```diff
@@ -0,0 +1,157 @@
+package kong
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/kong/semver/v4"
+)
+
+// Version represents a three or four digit version.
+type Version struct {
+	// version represents a three of four digit version using the forked kong/semver
+	// library.
+	version semver.Version
+	// str represents the original version string when creating Version.
+	str string
+}
+
+// Range represents a range of versions which can be use to validate if a Version is valid
+// for a Range.
+type Range func(Version) bool
+
+// NewVersion creates a new instance of a Version.
+func NewVersion(versionStr string) (Version, error) {
+	version, err := semver.Parse(versionStr)
+	if err != nil {
+		return Version{}, fmt.Errorf("unable to create version: %w", err)
```

Comment:
does the error from `semver` include the initial string? Otherwise it may be useful to include it in the error message.

Reply:
It does, here is an example of the error message using `3.0` as the `versionStr`

```
unable to create version: Invalid version: '3.0'
```

The `Invalid version: '3.0'` comes from the underlying `kong/semver` library.